lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/MavenCompletionParticipant.java

```diff
@@ -432,16 +434,34 @@ private CompletionItem toGAVCompletionItem(ArtifactInfo artifactInfo, ICompletio
 		if (artifactInfo.getDescription() != null) {
 			item.setDocumentation(artifactInfo.getDescription());
 		}
+		Range newReplaceRange;
+		try {
+			DOMDocument doc = request.getXMLDocument();
```

Comment:
In LemMinx we had a similar usecase when you try to open completion for element tag name inside a text node which doesn't start with `<`. See https://github.com/eclipse/lemminx/blob/0d812fb4acca54b9df17371e0bb8e8b99bcf5b73/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLCompletions.java#L755

If you see `getElementNameRangeAt `https://github.com/eclipse/lemminx/blob/0d812fb4acca54b9df17371e0bb8e8b99bcf5b73/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMDocument.java#L125 it consumes getWordRangeAt which use a regexp https://github.com/eclipse/lemminx/blob/0d812fb4acca54b9df17371e0bb8e8b99bcf5b73/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/commons/TextDocument.java#L95

I suggest that you write a regexp pattern for artifact id and you do 

`Range artifactIdRange = doc.getTextDocument(request.getOffset(), ARTIFACTID_PATTERN);`

(artifactdIdRange is your newReplaceRange )

Reply:
Thanks for pointing this out! I have switched to this method for the PR